src/main/java/de/tum/in/www1/artemis/domain/notification/GroupNotificationFactory.java

```diff
@@ -0,0 +1,162 @@
+package de.tum.in.www1.artemis.domain.notification;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.GroupNotificationType;
+import de.tum.in.www1.artemis.domain.enumeration.NotificationType;
+
+public class GroupNotificationFactory {
+
+    /**
+     * Creates an instance of GroupNotification based on the passed parameters.
+     *
+     * @param attachment for which a notification should be created
+     * @param author of the notification
+     * @param groupNotificationType user group type the notification should target
+     * @param notificationType type of the notification that should be created
+     * @return an instance of GroupNotification
+     */
+    public static GroupNotification createNotification(Attachment attachment, User author, GroupNotificationType groupNotificationType, NotificationType notificationType) {
+        String title, text;
+        if (notificationType == NotificationType.ATTACHMENT_CHANGE) {
+            title = "Attachment updated";
+            text = "Attachment \"" + attachment.getName() + "\" updated.";
+        }
+        else {
+            throw new UnsupportedOperationException("Unsupported NotificationType.");
+        }
+
+        Course course = attachment.getLecture().getCourse();
+        GroupNotification notification = new GroupNotification(course, title, text, author, groupNotificationType);
+
+        // Set target
+        notification.setTarget(notification.getAttachmentUpdated(attachment.getLecture()));
+
+        return notification;
+    }
+
+    /**
+     * Creates an instance of GroupNotification based on the passed parameters.
+     *
+     * @param exercise for which a notification should be created
+     * @param author of the notification
+     * @param groupNotificationType user group type the notification should target
+     * @param notificationType type of the notification that should be created
+     * @return an instance of GroupNotification
+     */
+    public static GroupNotification createNotification(Exercise exercise, User author, GroupNotificationType groupNotificationType, NotificationType notificationType) {
+        String title, text;
+        if (notificationType == NotificationType.EXERCISE_CREATED) {
```

Comment:
This seems like a good opportunity for a switch statement.

Reply:
Normally I prefer if-else over switch statements in Java but the new enhanced switch statements or really nice (multiple values possible, no break in every case needed). I will adjust my implementation :-)